### PR TITLE
Prepares virtual networking for UI

### PR DESCRIPTION
**Pull request by furkansahin**
This commit adds utility functions that are necessary for UI implementation.
Since these are model related changes, I am adding them with a separate commit.

### CODE REVIEW

File: prog/vm/nexus.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Prog::Vm::Nexus < Prog::Base
 
   def self.assemble(public_key, project_id, name: nil, size: "m5a.2x",
     unix_user: "ubi", location: "hetzner-hel1", boot_image: "ubuntu-jammy",
-    private_subnet_id: nil, nic_id: nil, storage_size_gib: 20, storage_encrypted: false)
+    private_subnet_ubid: nil, nic_ubid: nil, storage_size_gib: 20, storage_encrypted: false)
```
Comment (enescakir) [inline prog/vm/nexus.rb:14]: Should we get ubid as params or convert ubid to uuid at route file?

Comment (furkansahin) [reply prog/vm/nexus.rb:14]: This is a good question. I was thinking we would start using ubid everywhere but now that you say this. In Route, converting it and if it's not valid, simply failing there is a better idea. I will change it accordingly. 👍

File: prog/vm/nexus.rb [hunk 2]
```diff
@@ -41,9 +41,7 @@ def self.assemble(public_key, project_id, name: nil, size: "m5a.2x",
       # - If the user provided neither nic_id nor private_subnet_id, that's OK, we create both.
       nic = nil
       subnet = nil
-      if nic_id
-        nic = Nic[nic_id]
-        raise("Given nic doesn't exist with the id #{nic_id}") unless nic
```
Comment (enescakir) [inline prog/vm/nexus.rb:46]: I think we should keep existence check here too. For example, operator typed `nic_id` wrong and expects to use existing nic. But code will generate new nic. Same for `private_subnet_id` too

Comment (furkansahin) [reply prog/vm/nexus.rb]: I missed the operator error part, you're right. Added back.

**Review by enescakir — APPROVED** [review]

**Review by enescakir — APPROVED** [review]

**Pull request merged**